src/components/AppButton.vue

```diff
@@ -1,32 +1,34 @@
 <template>
-  <button :type="buttonType" :class="buttonIntent" @click="handleButtonClick">
+  <button :type="buttonType" :class="theme" @click="handleButtonClick">
     {{ title }}
   </button>
 </template>
 
 <script>
-import { CLOSE_APP_MODAL } from '@/store/mutation-types';
-import { mapMutations } from 'vuex';
-
 export default {
   name: 'AppButton',
   props: {
-    intent: { type: String, required: true },
+    isSubmit: {
```

Comment:
You don't need this `isSubmit` prop, you can just pass the type to the component directly and the root element will inherit that attribute 
https://vuejs.org/guide/components/attrs.html#attribute-inheritance